### PR TITLE
Add `cupy.conjugate` and make `cupy.conj` its alias

### LINKED ISSUE #2967: Rename `cupy.conj` to `cupy.conjugate`, add new `cupy.conj` alias
According to NumPy documentation, [`np.conj` is an alias for `np.conjugate`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.conj.html). However, CuPy implements [`cupy.conj`](https://docs-cupy.chainer.org/en/stable/reference/generated/cupy.conj.html) but not `cupy.conjugate`. Given that in NumPy the implementation is `np.conjugate`, I suggest renaming the current implementation to `cupy.conjugate` and make `cupy.conj` and alias for that.

Would there be any reason _not_ to do that?

### PR DESCRIPTION
**Pull request by toslunar**
Fix #2967.

### CODE REVIEW

**Comment by niboshi** [comment]
How about adding a test to ensure `cupy.conj is cupy.conjugate`?

**Comment by toslunar** [comment]
I didn't add the test in this PR just because there's no test for `cupy.abs is cupy.absolute`.

**Comment by niboshi** [comment]
If it's something that cupy needs to retain, we should have a test.

**Comment by toslunar** [comment]
> https://github.com/cupy/cupy/pull/2987#issuecomment-579056591

Done.

**Comment by niboshi** [comment]
This PR adds `cupy.ndarray.conjugate`. Can you add the corresponding test to `cupy_tests/core_tests/test_ndarray_complex_ops.py`?

**Comment by toslunar** [comment]
Do you mean `xp.ndarray.conj is not xp.ndarray.conjugate` should be tested?
```
% git show --format=raw
commit e27d263ad067f153370b8d3747fd2362c32d4da7
tree d5849beda18b96f6936d4d334aac5f8d1b864b5a
parent cc9c3f10d57de6adcf6d021be966df3f8cb8daf0
author Toshiki Kataoka <kataoka@preferred.jp> 1580275385 +0900
committer Toshiki Kataoka <kataoka@preferred.jp> 1580275385 +0900

    Test `conj` method is not `conjugate` method

diff --git a/tests/cupy_tests/test_init.py b/tests/cupy_tests/test_init.py
index ba4c19984..1b4a3708f 100644
--- a/tests/cupy_tests/test_init.py
+++ b/tests/cupy_tests/test_init.py
@@ -113,6 +113,12 @@ class TestAliases(unittest.TestCase):
         for xp in (numpy, cupy):
             assert xp.conj is xp.conjugate

+    def test_ndarray_conj_is_not_ndarray_conjugate(self):
+        for xp in (numpy, cupy):
+            assert xp.ndarray.conj is not xp.ndarray.conjugate
+            arr = xp.array([1, 2j])
+            assert arr.conj is not arr.conjugate
+

 # This is copied from chainer/testing/__init__.py, so should be replaced in
 # some way.
```

**Comment by niboshi** [comment]
@toslunar No, I meant `cupy.ndarray.conjugate` should be tested just the same way as the existing tests for `cupy.ndarray.conj` in `test_ndarray_complex_ops.py`.

**Comment by toslunar** [comment]
Then do you want something like the following?
```
% git show --format=raw
commit d41b9549e91852e83640d3661c81144727c2e4eb
tree eae0ccb21e486a4aa3553d29da5849b2de5651fe
parent cc9c3f10d57de6adcf6d021be966df3f8cb8daf0
author Toshiki Kataoka <kataoka@preferred.jp> 1580368699 +0900
committer Toshiki Kataoka <kataoka@preferred.jp> 1580368699 +0900

    Split test class

diff --git a/tests/cupy_tests/core_tests/test_ndarray_complex_ops.py b/tests/cupy_tests/core_tests/test_ndarray_complex_ops.py
index 395c26b34..373c2e682 100644
--- a/tests/cupy_tests/core_tests/test_ndarray_complex_ops.py
+++ b/tests/cupy_tests/core_tests/test_ndarray_complex_ops.py
@@ -22,6 +22,10 @@ class TestConj(unittest.TestCase):
         self.assertIs(x, y)
         return y

+
+@testing.gpu
+class TestConjugate(unittest.TestCase):
+
     @testing.for_all_dtypes()
     @testing.numpy_cupy_array_almost_equal()
     def test_conjugate(self, xp, dtype):
```

File: tests/cupy_tests/core_tests/test_ndarray_complex_ops.py [hunk 1]
```diff
@@ -22,6 +22,20 @@ def test_conj_pass(self, xp, dtype):
...
+        return x.conjugate()
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_array_almost_equal()
+    def test_conjugate_pass(self, xp, dtype):
+        x = testing.shaped_arange((2, 3), xp, dtype)
+        y = x.conjugate()
+        self.assertIs(x, y)
+        return y
+
```
Comment (niboshi) [inline tests/cupy_tests/core_tests/test_ndarray_complex_ops.py]: Sorry, it seems I overlooked this test for whatever reason.
This test is exactly what I meant.
Please forget my comment... :bowing_man:

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
:+1:

**Pull request merged**